src-extras/Database/LSMTree/Extras/Orphans.hs

```diff
@@ -35,3 +44,47 @@ instance SerialiseKey Word256 where
       return ba
 
   deserialiseKey = error "deserialiseKey: Word256" -- TODO
+
+instance SerialiseValue Word256 where
+  serialiseValue (Word256{word256hi, word256m1, word256m0, word256lo}) =
+    RB.RawBytes $ mkPrimVector 0 32 $ P.runByteArray $ do
+      ba <- P.newByteArray 32
+      P.writeByteArray ba 0 word256lo
+      P.writeByteArray ba 1 word256m0
+      P.writeByteArray ba 2 word256m1
+      P.writeByteArray ba 3 word256hi
+      return ba
+  deserialiseValue = error "deserialiseValue: Word256" -- TODO
+  deserialiseValueN = error "deserialiseValueN: Word256" -- TODO
+
+{-------------------------------------------------------------------------------
+  Orphans
+-------------------------------------------------------------------------------}
+
+deriving stock instance Generic (FS.HandleOS h)
+deriving anyclass instance NFData (FS.HandleOS h)
+
+deriving newtype instance NFData FS.BufferOffset
+
+deriving newtype instance NFData COff
+
+deriving instance Generic (FS.IOOp m h)
+deriving instance NFData h => NFData (FS.IOOp m h)
+
+deriving anyclass instance NFData FS.FsPath
+deriving instance NFData h => NFData (FS.Handle h)
+
+instance NFData (FS.HasFS m h) where
+  rnf x = x `seq` ()
+instance NFData (FS.HasBlockIO m h) where
+  rnf x = x `seq` ()
+
+deriving stock instance Generic (Run h)
+deriving anyclass instance NFData h => NFData (Run h)
+
+deriving newtype instance NFData RunFsPaths
+
+deriving newtype instance NFData BatchSize
+
+deriving stock instance Generic (BlobRef run)
+deriving anyclass instance NFData run => NFData (BlobRef run)
```

Comment:
Some types have their `NFData` instances where they are defined (e.g. `IndexCompact`), some here in an orphans module. Which approach do we generally want to go with?

Reply:
types in public interface should have `NFData` instances in the public interface. People will ask for them sooner or later anywa.

Reply:
@jorisdral Do you want to move these instances to the type definitions as part of this PR then?

Reply:
See #193